src/librbd/cache/pwl/ssd/WriteLog.cc

```diff
@@ -288,24 +270,30 @@ void WriteLog<I>::load_existing_entries(pwl::DeferredContexts &later) {
 template <typename I>
 bool WriteLog<I>::alloc_resources(C_BlockIORequestT *req) {
   bool alloc_succeeds = true;
+  bool no_space = false;
   uint64_t bytes_allocated = 0;
   uint64_t bytes_cached = 0;
   uint64_t bytes_dirtied = 0;
   uint64_t num_lanes = 0;
   uint64_t num_unpublished_reserves = 0;
   uint64_t num_log_entries = 0;
 
-  // Setup buffer, and get all the number of required resources
   req->setup_buffer_resources(&bytes_cached, &bytes_dirtied, &bytes_allocated,
                               &num_lanes, &num_log_entries,
                               &num_unpublished_reserves);
 
   bytes_allocated += num_log_entries * MIN_WRITE_ALLOC_SSD_SIZE;
 
-  alloc_succeeds = this->check_allocation(req, bytes_cached, bytes_dirtied,
-                                          bytes_allocated, num_lanes,
-                                          num_log_entries,
-                                          num_unpublished_reserves);
+  this->check_allocation(req, bytes_cached, bytes_dirtied,
+                         bytes_allocated, num_lanes,
+                         num_log_entries,
+                         num_unpublished_reserves,
+                         this->m_bytes_allocated_cap,
+                         alloc_succeeds, no_space);
+  this->m_bytes_allocated += bytes_allocated;
+  this->m_bytes_cached += bytes_cached;
+  this->m_bytes_dirty += bytes_dirtied;
```

Comment:
Did you mean to bump these counters unconditionally, even if `alloc_succeeds` is false?  This messes up the cache pretty quickly and I/O just hangs.

Reply:
Ugh, I'm sorry. No, I didn't intend to remove the condition of `alloc_succeeds`. I'll fix it.

Reply:
It's fixed (or rather avoided altogether) in my patch.  Did you see the paste?

I prefer that approach because it minimizes the risk of regression for rwl mode -- only ssd mode is patched and there is a lot less code churn.

Reply:
I did see the paste. You're right that there's less code churn but I was wondering if it would be more clear to remove them altogether in `ssd` code as that would make it more clear that they do not belong in `ssd` logic - I think this would help future code reviewers/maintainers?

Reply:
Well, your patch left those fields in `AbstractWriteLog`, so they weren't removed altogether -- `ssd/WriteLog` could still access them.  Doing it properly would cause even more code churn and I'd like to avoid that until everything is stable.

Reply:
> Well, your patch left those fields in `AbstractWriteLog`, so they weren't removed altogether -- `ssd/WriteLog` could still access them. 

Right. I could correct those, not saying this PR should be merged as is. It's a thought I had when I looked at the paste you shared
>Doing it properly would cause even more code churn and I'd like to avoid that until everything is stable.

okay